### PR TITLE
feat: Adds ability to expanded reply box

### LINKED ISSUE #508: feature: Pop out reply box to write long messages
- Add expand icon near send button in reply box
- On clicking expand icon, show the reply box in a modal
- Add minimize icon in reply box modal
- User should be able to close the model by clicking on it

**Design**

Simulator Screen Shot - iPhone 8 Plus - 2022-05-05 at 12 40

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description 

This PR adds the ability to increase the height of the input field.

Fixes #508

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

1. Open a conversation.
2. Now you can see an expand button on the bottom left side.

**Screencast**

https://user-images.githubusercontent.com/64252451/168733170-6a950f80-7215-487c-940b-bdd2d0fe943f.mov

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have tested in both Android and iOS platform
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/screens/ChatScreen/components/ReplyBox.js [hunk 1]
```diff
@@ -108,6 +109,30 @@ const ReplyBox = ({ eva: { theme, style }, conversationId, cannedResponses }) =>
...
+    if (isExpanded) {
+      return {
+        height: 400,
+      };
+    } else {
+      return {
+        height: 40,
+      };
+    }
+  };
```
Comment (muhsin-k) [inline src/screens/ChatScreen/components/ReplyBox.js:122]:
```suggestion
 const inputFieldHeight = () => (isExpanded ? { height: 400 } : { height: 40 });
```

File: src/screens/ChatScreen/components/ReplyBox.js [hunk 2]
```diff
@@ -108,6 +109,30 @@ const ReplyBox = ({ eva: { theme, style }, conversationId, cannedResponses }) =>
...
+
+  const expandedInputButtonIcons = () => {
+    if (isExpanded) {
+      return 'collapse-outline';
+    } else {
+      return 'expand-outline';
+    }
+  };
```
Comment (muhsin-k) [inline src/screens/ChatScreen/components/ReplyBox.js:130]:
```suggestion
const expandedInputButtonIcon = () => (isExpanded ? 'collapse-outline' : 'expand-outline');
```

File: src/screens/ChatScreen/components/ReplyBox.js [hunk 3]
```diff
@@ -185,6 +210,15 @@ const ReplyBox = ({ eva: { theme, style }, conversationId, cannedResponses }) =>
                 onPress={togglePrivateMode}
               />
             </View>
+            <View style={style.expandButton}>
+              <Icon
+                name={expandedInputButtonIcons()}
+                width={32}
```
Comment (muhsin-k) [inline src/screens/ChatScreen/components/ReplyBox.js:216]: Seems like icons are big now. Please change icon size to 24 (Attachment, Private note and Expanded)

**Review by muhsin-k — COMMENTED** [review]
@iamsivin Nice work. Please see the inline comments

File: src/screens/ChatScreen/components/ReplyBox.js (hunk 3 above)
Comment (iamsivin) [reply src/screens/ChatScreen/components/ReplyBox.js]: image

@muhsin-k Does the send button size ok?

Comment (muhsin-k) [reply src/screens/ChatScreen/components/ReplyBox.js]: @iamsivin Keeps send button icon same

Comment (iamsivin) [reply src/screens/ChatScreen/components/ReplyBox.js]: @muhsin-k  I think the same size looks better. 

image

**Comment by pranavrajs** [comment]
@iamsivin @muhsin-k Can we expand the editor as and when we type?

**Comment by iamsivin** [comment]
> @iamsivin @muhsin-k Can we expand the editor as and when we type?

Sure, done https://github.com/chatwoot/chatwoot-mobile-app/pull/523/commits/860655271f7e0243dba2a84af077d5564dec3e9c

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**